### PR TITLE
more kanban tests (pager ones)

### PR DESCRIPTION
**Pull request by jpp-odoo**

### CODE REVIEW

File: addons/web/static/src/views/kanban/kanban_controller.js [hunk 1]
```diff
@@ -71,7 +71,7 @@ export class KanbanController extends Component {
                     limit: limit,
                     total: count,
                     onUpdate: async ({ offset, limit }) => {
-                        await this.model.load({ offset, limit });
+                        await this.model.root.load({ offset, limit });
```
Comment (aab-odoo) [inline addons/web/static/src/views/kanban/kanban_controller.js:74]: why not model.load?

File: addons/web/static/src/views/relational_model/dynamic_record_list.js [hunk 2]
```diff
@@ -17,7 +17,7 @@ export class DynamicRecordList extends DynamicList {
                     data: r,
                 })
         );
-        this.hasLimitedCount = false;
+        this.countLimit = params.countLimit || this.constructor.WEB_SEARCH_READ_COUNT_LIMIT;
```
Comment (aab-odoo) [inline addons/web/static/src/views/relational_model/dynamic_record_list.js]: what about initial read?

File: addons/web/static/src/views/relational_model/dynamic_record_list.js [hunk 3]
```diff
@@ -37,15 +37,39 @@ export class DynamicRecordList extends DynamicList {
...
     }
 
+    async load(params = {}) {
+        this.limit = params.limit === undefined ? this.limit : params.limit;
+        this.offset = params.offset === undefined ? this.offset : params.offset;
+        await this._load();
+    }
+
+    async deleteRecords(records) {
+        await Promise.all(records.map((record) => record.delete()));
```
Comment (aab-odoo) [inline addons/web/static/src/views/relational_model/dynamic_record_list.js]: one rpc by record? Also, delete does other stuff that we probably don't want here

File: addons/web/static/src/views/relational_model/dynamic_record_list.js [hunk 4]
```diff
@@ -37,15 +37,39 @@ export class DynamicRecordList extends DynamicList {
...
 
+    _removeRecords(records) {
+        this.records = this.records.filter((r) => !records.includes(r));
+        this._updateCount(this.records);
+        this._adjustOffset();
+    }
+
+    _adjustOffset() {
+        if (this.offset && !this.records.length) {
+            this.offset = Math.max(this.offset - this.limit, 0);
```
Comment (aab-odoo) [inline addons/web/static/src/views/relational_model/dynamic_record_list.js]: updates the model in place before loading -> multiple renderings + incorrect state for a while

File: addons/web/static/src/views/relational_model/dynamic_record_list.js [hunk 5]
```diff
@@ -37,15 +37,39 @@ export class DynamicRecordList extends DynamicList {
...
+            this._load();
```
Comment (aab-odoo) [inline addons/web/static/src/views/relational_model/dynamic_record_list.js]: wait?

File: addons/web/static/src/views/relational_model/group.js [hunk 6]
```diff
@@ -58,6 +58,11 @@ export class Group extends DataPoint {
         this.isFolded = !this.isFolded;
     }
 
+    async deleteRecords(records) {
+        this.count = this.count - records.length;
```
Comment (aab-odoo) [inline addons/web/static/src/views/relational_model/group.js:63]: always ensure the model is mutated only once, in the same tick

File: addons/web/static/src/views/kanban/kanban_controller.js (hunk 1 above)
Comment (jpp-odoo) [reply addons/web/static/src/views/kanban/kanban_controller.js:74]: OK

File: addons/web/static/src/views/relational_model/dynamic_record_list.js (hunk 2 above)
Comment (jpp-odoo) [reply addons/web/static/src/views/relational_model/dynamic_record_list.js]: to the model !

File: addons/web/static/src/views/relational_model/dynamic_record_list.js (hunk 3 above)
Comment (jpp-odoo) [reply addons/web/static/src/views/relational_model/dynamic_record_list.js]: rpc unlink all ids (and not one)

File: addons/web/static/src/views/relational_model/dynamic_record_list.js (hunk 4 above)
Comment (jpp-odoo) [reply addons/web/static/src/views/relational_model/dynamic_record_list.js]:
```suggestion
            const offset = Math.max(this.offset - this.limit, 0);
```

File: addons/web/static/src/views/relational_model/dynamic_record_list.js (hunk 5 above)
Comment (jpp-odoo) [reply addons/web/static/src/views/relational_model/dynamic_record_list.js]:
```suggestion
           await this._load({offset});
```

File: addons/web/static/src/views/relational_model/dynamic_record_list.js [hunk 7]
```diff
@@ -55,13 +79,20 @@ export class DynamicRecordList extends DynamicList {
     async _load() {
```
Comment (jpp-odoo) [inline addons/web/static/src/views/relational_model/dynamic_record_list.js]:
```suggestion
    async _load(params) {
```

File: addons/web/static/src/views/relational_model/dynamic_record_list.js [hunk 8]
```diff
@@ -37,15 +37,39 @@ export class DynamicRecordList extends DynamicList {
         return this.count;
     }
 
+    async load(params = {}) {
+        this.limit = params.limit === undefined ? this.limit : params.limit;
```
Comment (jpp-odoo) [inline addons/web/static/src/views/relational_model/dynamic_record_list.js]: params to the _load

**Pull request merged**